packages/webapp/pages/onboarding.tsx

```diff
@@ -134,23 +210,31 @@ export function OnboardPage(): ReactElement {
       return;
     }
 
-    if (user || onboardingV2 === OnboardingV2.Control) {
+    if (user || (onboardingV2 === OnboardingV2.Control && !isOnboardingV3)) {
       router.push('/');
       return;
     }
 
     trackEvent({
       event_name: AnalyticsEvent.Impression,
       target_type: TargetType.MyFeedModal,
-      target_id: onboardingV2,
+      target_id: isOnboardingV3 ? onboardingV3 : onboardingV2,
       extra: JSON.stringify({
         origin: OnboardingMode.Wall,
         steps: [OnboardingStep.Topics],
         mandating_categories: 0,
       }),
     });
     isTracked.current = true;
-  }, [trackEvent, isPageReady, onboardingV2, router, user]);
+  }, [
+    trackEvent,
+    isPageReady,
+    onboardingV2,
+    router,
```

Comment:
I know `router` was here before but it has an unstable reference in Next 12.x so might be better to just remove it for now until we upgrade.

Reply:
Ref: https://github.com/vercel/next.js/issues/18127

Reply:
Removed and added eslint ignore